esp/esdllib/esdl_def.hpp

```diff
@@ -230,4 +233,144 @@ esdl_decl void releaseEsdlDefinition(const char *esdl_ns=NULL);
 esdl_decl void initEsdlTypeList();
 esdl_decl EsdlBasicElementType esdlSimpleType(const char *type);
 
+interface IEsdlDefReporter : extends IInterface
+{
+    using Flags = uint64_t;
+    static const Flags ReportDisaster  = 1 << 0;
+    static const Flags ReportAError    = 1 << 1;
+    static const Flags ReportIError    = 1 << 2;
+    static const Flags ReportOError    = 1 << 3;
+    static const Flags ReportUError    = 1 << 4;
+    static const Flags ReportIWarning  = 1 << 5;
+    static const Flags ReportOWarning  = 1 << 6;
+    static const Flags ReportUWarning  = 1 << 7;
+    static const Flags ReportDProgress = 1 << 8;
+    static const Flags ReportOProgress = 1 << 9;
+    static const Flags ReportUProgress = 1 << 10;
+    static const Flags ReportDInfo     = 1 << 11;
+    static const Flags ReportOInfo     = 1 << 12;
+    static const Flags ReportUInfo     = 1 << 13;
+    static const Flags ReportStats     = 1 << 14;
+    static const Flags ReportCategoryMask =
+            ReportDisaster |
+            ReportAError | ReportIError | ReportOError | ReportUError |
+            ReportIWarning | ReportOWarning | ReportUWarning |
+            ReportDProgress | ReportOProgress | ReportUProgress |
+            ReportDInfo | ReportOInfo | ReportUInfo |
+            ReportStats;
+
+    static const Flags ReportDefinition = Flags(1) << 63;
+    static const Flags ReportService    = Flags(1) << 62;
+    static const Flags ReportMethod     = Flags(1) << 61;
+    static const Flags ReportComponentMask = Flags(-1) & ~ReportCategoryMask;
```

Comment:
Could we use UINT64_MAX here to be more clear?

Reply:
Yes